File: web/plugins/gcd.js
import { WebPlugin } from "../engine/plugin.js";

export class GcdPlugin extends WebPlugin {
    constructor() {
        super("最大公約数");
    }

    async execute(line, context) {
        const match = line.match(/^最大公約数\s+(.+)\s+(.+)$/);
        if (!match) return;

        let a = Math.abs(Number(
            context.parser.parseValue(match[1], context.runtime)
        ));

        let b = Math.abs(Number(
            context.parser.parseValue(match[2], context.runtime)
        ));

        while (b !== 0) {
            [a, b] = [b, a % b];
        }

        context.runtime.write(a);
    }
}

File: pc/plugins/gcd.py
import re
from core.plugin import Plugin


class GcdPlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'最大公約数\s+(.+)\s+(.+)', line)
        if not match:
            return None

        a = match.group(1)
        b = match.group(2)

        return f"print(__import__('math').gcd({a}, {b}))"